crates/core/tedge_actors/src/actors.rs

```diff
@@ -207,19 +202,19 @@ pub mod tests {
     use futures::StreamExt;
     use tokio::spawn;
 
-    struct Echo;
+    struct Echo {
+        messages: SimpleMessageBox<String, String>,
+    }
 
     #[async_trait]
     impl Actor for Echo {
-        type MessageBox = SimpleMessageBox<String, String>;
-
         fn name(&self) -> &str {
             "Echo"
         }
 
-        async fn run(mut self, mut messages: Self::MessageBox) -> Result<(), RuntimeError> {
-            while let Some(message) = messages.recv().await {
-                messages.send(message).await?
+        async fn run(mut self) -> Result<(), RuntimeError> {
+            while let Some(message) = self.messages.recv().await {
+                self.messages.send(message).await?
```

Comment:
This is good, when I was using the `actor` framework to implement the `health` actor, I was really confused that why the `message box` is sending and receiving. It must be the actor which has to send and receive, now it's clear.